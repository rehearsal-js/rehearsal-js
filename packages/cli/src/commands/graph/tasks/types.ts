export interface GraphWorkerOptions {
  rootPath: string;
  srcPath: string;
  ignore: string[];
  serviceMap: Record<string, string>;
  output?: string;
}

type MessageResponse = { type: 'message'; content: string };
type FilesResponse = { type: 'files'; content: string[] };
export type Response = MessageResponse | FilesResponse;
