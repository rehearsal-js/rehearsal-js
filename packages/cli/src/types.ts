import { MigrationStrategy } from '@rehearsal/migration-graph';

import { UserConfig } from './userConfig';
import { State } from './helpers/state';

export type CliCommand = 'upgrade' | 'migrate';
export type Formats = 'sarif' | 'json' | 'sonarqube' | 'md';

export type MigrateCommandOptions = {
  basePath: string;
  entrypoint: string;
  files: string;
  format: Formats[];
  outputPath: string;
  verbose: boolean | undefined;
  userConfig: string | undefined;
  interactive: boolean | undefined;
};

export type MigrateCommandContext = {
  skip: boolean;
  userConfig: UserConfig | undefined;
  strategy: MigrationStrategy;
  sourceFilesWithAbsolutePath: string[];
  sourceFilesWithRelativePath: string[];
  input: unknown;
  targetPackagePath: string;
  state: State;
};

export type UpgradeCommandContext = {
  tsVersion: string;
  latestAvailableBuild: string;
  currentTSVersion: string;
  skip: boolean;
};

export type UpgradeCommandOptions = {
  build: string;
  basePath: string;
  format: Formats[];
  outputPath: string;
  tsVersion: string;
  dryRun: boolean;
};

export type DependencyConfig = {
  dependencies?: string[];
  devDependencies?: string[];
};

export type SetupConfigCommand = {
  command: string;
  args: string[];
};

export type SetupConfig = {
  ts?: SetupConfigCommand;
  lint?: SetupConfigCommand;
};

export type CustomCommandConfig = {
  install?: DependencyConfig;
  setup?: SetupConfig;
};

export type CustomConfig = {
  upgrade?: CustomCommandConfig;
  migrate?: CustomCommandConfig;
};

export type PackageSelection = {
  name: string;
  path: string;
};

export type MenuMap = {
  [key: string]: string;
};

export type MigrationSummary = {
  totalErrorCount: number;
  hintAddedCount: number;
};
export type TSConfig = {
  compilerOptions: {
    strict: boolean;
  };
};

export type ScriptMap = {
  [key: string]: string;
};
