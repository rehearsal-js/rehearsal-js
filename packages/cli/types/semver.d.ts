declare module 'semver' {
  const semver: {
    valid: (input: string) => boolean;
  };
  export = semver;
}
