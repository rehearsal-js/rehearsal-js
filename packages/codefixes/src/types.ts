import {
  CodeFixAction,
  DiagnosticWithLocation,
  FileTextChanges,
  LanguageService,
  Node,
  Program,
  TypeChecker,
} from 'typescript';

export interface DiagnosticWithContext extends DiagnosticWithLocation {
  service: LanguageService;
  program: Program;
  checker: TypeChecker;
  node?: Node;
}

export interface CodeHintList {
  [key: number]: CodeHintListElement;
}

export interface CodeHintListElement {
  hint: string;
  helpUrl?: string;
  hints?: CodeHint[];
}

export interface CodeHint {
  when: (n: Node, p: Program, c: TypeChecker) => boolean;
  hint: string;
}

export interface CodeFixCollection {
  getFixForDiagnostic(diagnostic: DiagnosticWithContext): CodeFixAction | undefined;
}

export type CodeFixKind = 'insert' | 'delete' | 'replace';

export interface FixedFile {
  fileName: string;
  newCode?: string;
  oldCode?: string;
  codeFixAction?: CodeFixKind;
  location: {
    startLine: number;
    startColumn: number;
    endLine: number;
    endColumn: number;
  };
}

export interface CodeFix {
  getCodeAction: (diagnostic: DiagnosticWithContext) => CodeFixAction | undefined;
}

export function createCodeFixAction(
  fixName: string,
  changes: FileTextChanges[],
  description: string
): CodeFixAction {
  return {
    fixName,
    description,
    changes,
  };
}
