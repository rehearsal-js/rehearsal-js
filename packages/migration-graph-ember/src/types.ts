/* eslint-disable @typescript-eslint/no-explicit-any */
import type { Package } from '@rehearsal/migration-graph-shared';
import type { EmberAddonPackage } from './entities/ember-addon-package.js';
import type { EmberAppPackage } from './entities/ember-app-package.js';

export type EmberProjectPackage = Package | EmberAddonPackage | EmberAppPackage;
