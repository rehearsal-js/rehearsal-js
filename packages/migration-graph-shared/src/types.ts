import type { Package } from './entities/package.js';

export type UniqueNode = {
  key: string;
  synthetic?: boolean;
};

export type PackageNode = UniqueNode & {
  pkg: Package;
  converted?: boolean;
};

export type ModuleNode = UniqueNode & {
  path: string;
  parsed?: boolean;
};
