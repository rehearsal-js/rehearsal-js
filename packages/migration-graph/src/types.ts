import { Package } from '@rehearsal/migration-graph-shared';

import { Graph } from './utils/graph';

export type UniqueGraphNode = {
  key: string;
};

export type PackageNode = UniqueGraphNode & {
  pkg: Package;
  converted?: boolean;
  modules: Graph<ModuleNode>;
};

export type ModuleNode = UniqueGraphNode & {
  path: string;
  parsed?: boolean;
  meta?: unknown;
};
