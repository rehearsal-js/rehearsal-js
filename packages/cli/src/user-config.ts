import { resolve } from 'node:path';
import { addDep } from '@rehearsal/utils';
import { execa } from 'execa';
import { readJSONSync } from 'fs-extra/esm';
import type {
  MigrateCommandConfig,
  UpgradeCommandConfig,
  CliCommand,
  CustomConfig,
} from './types.js';

// Storage and runner for user custom cli config
export class UserConfig {
  public basePath: string;
  public config?: MigrateCommandConfig | UpgradeCommandConfig;

  constructor(basePath: string, command: CliCommand) {
    const config: CustomConfig = readJSONSync(resolve(basePath, 'rehearsal-config.json'));

    this.config = config[command];

    this.basePath = basePath;
  }

  public get hasDependencies(): boolean {
    return (
      !!this.config?.install &&
      (!!this.config?.install?.dependencies || !!this.config?.install?.devDependencies)
    );
  }

  public get dependencies(): string[] {
    if (this.config && this.config.install) {
      const { dependencies } = this.config.install;
      if (dependencies && dependencies.length) {
        return dependencies;
      }
    }
    return [];
  }

  public get devDependencies(): string[] {
    if (this.config && this.config.install) {
      const { devDependencies } = this.config.install;
      if (devDependencies && devDependencies.length) {
        return devDependencies;
      }
    }
    return [];
  }

  public get hasPostInstallHook(): boolean {
    return !!this.config?.postInstall;
  }

  public get hasTsSetup(): boolean {
    return !!this.config?.setup?.ts;
  }

  public get hasPostTsSetupHook(): boolean {
    return !!this.config?.setup?.postTsSetup;
  }

  public get hasLintSetup(): boolean {
    return !!this.config?.setup?.lint;
  }

  public get hasPostLintSetup(): boolean {
    return !!this.config?.setup?.postLintSetup;
  }

  async install(): Promise<void> {
    if (this.config && this.config.install) {
      const { dependencies, devDependencies } = this.config.install;
      if (dependencies && dependencies.length) {
        await addDep(dependencies, false, { cwd: this.basePath });
      }

      if (devDependencies && devDependencies.length) {
        await addDep(devDependencies, true, { cwd: this.basePath });
      }
    }
  }

  async postInstall(): Promise<void> {
    if (this.config && this.config.postInstall) {
      if (Array.isArray(this.config.postInstall)) {
        for (const { command, args } of this.config.postInstall) {
          await execa(command, args, { cwd: this.basePath });
        }
      } else {
        const { command, args } = this.config.postInstall;
        await execa(command, args, { cwd: this.basePath });
      }
    }
  }

  // TODO: how to make setup tasks more generic instead of separated functions?

  // TODO: it doesn't make sense for migrate and upgrade sharing the same setup tasks,
  // What does upgrade command need for custom config?
  async tsSetup(): Promise<void> {
    if (this.config?.setup?.ts) {
      const { ts: tsSetup } = this.config.setup;
      if (Array.isArray(tsSetup)) {
        for (const { command, args } of tsSetup) {
          await execa(command, args, { cwd: this.basePath });
        }
      } else {
        const { command, args } = tsSetup;
        await execa(command, args, { cwd: this.basePath });
      }
    }
  }

  async postTsSetup(): Promise<void> {
    if (this.config?.setup?.postTsSetup) {
      const { postTsSetup } = this.config.setup;
      if (Array.isArray(postTsSetup)) {
        for (const { command, args } of postTsSetup) {
          await execa(command, args, { cwd: this.basePath });
        }
      } else {
        const { command, args } = postTsSetup;
        await execa(command, args, { cwd: this.basePath });
      }
    }
  }

  async lintSetup(): Promise<void> {
    if (this.config?.setup?.lint) {
      const { lint: lintSetup } = this.config.setup;
      if (Array.isArray(lintSetup)) {
        for (const { command, args } of lintSetup) {
          await execa(command, args, { cwd: this.basePath });
        }
      } else {
        const { command, args } = lintSetup;
        await execa(command, args, { cwd: this.basePath });
      }
    }
  }

  async postLintSetup(): Promise<void> {
    if (this.config?.setup?.postLintSetup) {
      const { postLintSetup } = this.config.setup;
      if (Array.isArray(postLintSetup)) {
        for (const { command, args } of postLintSetup) {
          await execa(command, args, { cwd: this.basePath });
        }
      } else {
        const { command, args } = postLintSetup;
        await execa(command, args, { cwd: this.basePath });
      }
    }
  }

  get exclude(): string[] | undefined {
    return (this.config as MigrateCommandConfig)?.exclude;
  }

  get include(): string[] | undefined {
    return (this.config as MigrateCommandConfig)?.include;
  }
}
