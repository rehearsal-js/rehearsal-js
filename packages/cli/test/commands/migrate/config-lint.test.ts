import { resolve } from 'node:path';
import { readdirSync, writeFileSync } from 'node:fs';
import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import { writeJSONSync } from 'fs-extra/esm';
import { cosmiconfigSync } from 'cosmiconfig';

import { depInstallTask, lintConfigTask } from '../../../src/commands/migrate/tasks/index.js';
import { prepareTmpDir, listrTaskRunner, createMigrateOptions } from '../../test-helpers/index.js';
import { CustomConfig } from '../../../src/types.js';
import { UserConfig } from '../../../src/user-config.js';

function createUserConfig(basePath: string, config: CustomConfig): void {
  const configPath = resolve(basePath, 'rehearsal-config.json');
  writeJSONSync(configPath, config);
}

describe('Task: config-lint', () => {
  let basePath = '';
  let output = '';
  vi.spyOn(console, 'info').mockImplementation((chunk) => {
    output += `${chunk}\n`;
  });
  vi.spyOn(console, 'log').mockImplementation((chunk) => {
    output += `${chunk}\n`;
  });

  let explorerSync;

  beforeEach(() => {
    output = '';
    basePath = prepareTmpDir('initialization');
  });

  afterEach(() => {
    vi.clearAllMocks();
    explorerSync = null;
  });

  test('create .eslintrc.js if not existed', async () => {
    const options = createMigrateOptions(basePath);
    // lint task requires dependencies installed first
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc.js');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc.js');
    expect(output).matchSnapshot();
  });

  test('extends .eslintrc.js if existed', async () => {
    // prepare old .eslintrc.js
    const oldConfig = `
    module.exports = {extends: []};
  `;
    writeFileSync(resolve(basePath, '.eslintrc.js'), oldConfig);

    const options = createMigrateOptions(basePath);
    // lint task requires dependencies installed first
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc.js');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc.js');
    // Do not use snapshot here since there is absolute path in output
    expect(output).toContain('extending Rehearsal default eslint-related config');

    /* eslint-disable-next-line @typescript-eslint/no-var-requires */
    const newConfig = require(resolve(basePath, '.eslintrc.js')) as { extends: string[] };
    expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc.js']);
  });

  test('skip if .eslintrc.js exists with valid extends', async () => {
    // prepare old .eslintrc.js and .rehearsal-eslintrc.js
    const oldConfig = `
    module.exports = {extends: ["./.rehearsal-eslintrc.js"]};
  `;
    writeFileSync(resolve(basePath, '.eslintrc.js'), oldConfig);
    writeFileSync(resolve(basePath, '.rehearsal-eslintrc.js'), '');

    const options = createMigrateOptions(basePath);
    // this validation does not need depInstallTask
    const tasks = [lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(output).toContain('[SKIPPED] Create eslint config');
  });

  test('extends .eslintrc if existed', async () => {
    const oldConfig = `
    {"extends": []}
  `;
    writeFileSync(resolve(basePath, '.eslintrc'), oldConfig);

    const options = createMigrateOptions(basePath);
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc');
    // Do not use snapshot here since there is absolute path in output
    expect(output).toContain('extending Rehearsal default eslint-related config');

    explorerSync = cosmiconfigSync('');
    const loaded = explorerSync.load(resolve(basePath, '.eslintrc'));
    const newConfig = loaded?.config as { extends: string[] };
    expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc']);
  });

  test('skip if .eslintrc exists with valid extends', async () => {
    // prepare old .eslintrc and .rehearsal-eslintrc.js
    const oldConfig = `
    {extends: ["./.rehearsal-eslintrc.js"]}
  `;
    writeFileSync(resolve(basePath, '.eslintrc'), oldConfig);
    writeFileSync(resolve(basePath, '.rehearsal-eslintrc.js'), '');

    const options = createMigrateOptions(basePath);
    // this validation does not need depInstallTask
    const tasks = [lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(output).toContain('[SKIPPED] Create eslint config');
  });

  test('extends .eslintrc.json if existed', async () => {
    const oldConfig = `
    {"extends": []}
  `;
    writeFileSync(resolve(basePath, '.eslintrc.json'), oldConfig);

    const options = createMigrateOptions(basePath);
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc.json');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc.json');
    // Do not use snapshot here since there is absolute path in output
    expect(output).toContain('extending Rehearsal default eslint-related config');

    explorerSync = cosmiconfigSync('');
    const loaded = explorerSync.load(resolve(basePath, '.eslintrc.json'));
    const newConfig = loaded?.config as { extends: string[] };
    expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc.json']);
  });

  test('skip if .eslintrc.json exists with valid extends', async () => {
    // prepare old .eslintrc.json and .rehearsal-eslintrc.js
    const oldConfig = `
    {"extends": ["./.rehearsal-eslintrc.js"]}
  `;
    writeFileSync(resolve(basePath, '.eslintrc.json'), oldConfig);
    writeFileSync(resolve(basePath, '.rehearsal-eslintrc.js'), '');

    const options = createMigrateOptions(basePath);
    // this validation does not need depInstallTask
    const tasks = [lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(output).toContain('[SKIPPED] Create eslint config');
  });

  test('extends .eslintrc.yml if existed', async () => {
    const oldConfig = `
    extends: []
  `;
    writeFileSync(resolve(basePath, '.eslintrc.yml'), oldConfig);

    const options = createMigrateOptions(basePath);
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc.yml');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc.yml');
    // Do not use snapshot here since there is absolute path in output
    expect(output).toContain('extending Rehearsal default eslint-related config');

    explorerSync = cosmiconfigSync('');
    const loaded = explorerSync.load(resolve(basePath, '.eslintrc.yml'));
    const config = loaded?.config as { extends: string[] };

    /* eslint-disable-next-line @typescript-eslint/no-var-requires */
    expect(config.extends).toStrictEqual(['./.rehearsal-eslintrc.yml']);
  });

  test('skip if .eslintrc.yml exists with valid extends', async () => {
    // prepare old .eslintrc.yml and .rehearsal-eslintrc.js
    const oldConfig = `
    extends: ["./.rehearsal-eslintrc.js"]
  `;
    writeFileSync(resolve(basePath, '.eslintrc.yml'), oldConfig);
    writeFileSync(resolve(basePath, '.rehearsal-eslintrc.js'), '');

    const options = createMigrateOptions(basePath);
    // this validation does not need depInstallTask
    const tasks = [lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(output).toContain('[SKIPPED] Create eslint config');
  });

  test('extends .eslintrc.yaml if existed', async () => {
    const oldConfig = `
    extends: []
  `;
    writeFileSync(resolve(basePath, '.eslintrc.yaml'), oldConfig);

    const options = createMigrateOptions(basePath);
    const tasks = [depInstallTask(options), lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(readdirSync(basePath)).toContain('.eslintrc.yaml');
    expect(readdirSync(basePath)).toContain('.rehearsal-eslintrc.yaml');
    // Do not use snapshot here since there is absolute path in output
    expect(output).toContain('extending Rehearsal default eslint-related config');

    explorerSync = cosmiconfigSync('');
    const loaded = explorerSync.load(resolve(basePath, '.eslintrc.yaml'));
    const config = loaded?.config as { extends: string[] };

    /* eslint-disable-next-line @typescript-eslint/no-var-requires */
    expect(config.extends).toStrictEqual(['./.rehearsal-eslintrc.yaml']);
  });

  test('skip if .eslintrc.yaml exists with valid extends', async () => {
    // prepare old .eslintrc.yaml and .rehearsal-eslintrc.js
    const oldConfig = `
    extends: ["./.rehearsal-eslintrc.js"]
  `;
    writeFileSync(resolve(basePath, '.eslintrc.yaml'), oldConfig);
    writeFileSync(resolve(basePath, '.rehearsal-eslintrc.js'), '');

    const options = createMigrateOptions(basePath);
    // this validation does not need depInstallTask
    const tasks = [lintConfigTask(options)];
    await listrTaskRunner(tasks);

    expect(output).toContain('[SKIPPED] Create eslint config');
  });

  test('run custom config command with user config provided', async () => {
    createUserConfig(basePath, {
      migrate: {
        setup: {
          lint: { command: 'touch', args: ['custom-lint-config-script'] },
        },
      },
    });

    const options = createMigrateOptions(basePath, { userConfig: 'rehearsal-config.json' });
    const userConfig = new UserConfig(basePath, 'rehearsal-config.json', 'migrate');
    const tasks = [depInstallTask(options), lintConfigTask(options, { userConfig })];
    await listrTaskRunner(tasks);

    // This proves the custom command works
    expect(readdirSync(basePath)).toContain('custom-lint-config-script');
    expect(output).toMatchSnapshot();
  });

  test('postLintSetup hook from user config', async () => {
    createUserConfig(basePath, {
      migrate: {
        setup: {
          lint: { command: 'touch', args: ['custom-lint-config-script'] },
          postLintSetup: { command: 'mv', args: ['custom-lint-config-script', 'foo'] },
        },
      },
    });

    const options = createMigrateOptions(basePath, { userConfig: 'rehearsal-config.json' });
    const userConfig = new UserConfig(basePath, 'rehearsal-config.json', 'migrate');
    const tasks = [depInstallTask(options), lintConfigTask(options, { userConfig })];
    await listrTaskRunner(tasks);

    // This proves the custom command and hook works
    expect(readdirSync(basePath)).toContain('foo');
    expect(readdirSync(basePath)).not.toContain('custom-lint-config-script');
    expect(output).toMatchSnapshot();
  });

  test('skip if custom config and .eslintrc.js exist', async () => {
    // prepare old .eslintrc.js
    const oldConfig = `
    module.exports = {extends: ["./.rehearsal-eslintrc.js"]};
  `;
    writeFileSync(resolve(basePath, '.eslintrc.js'), oldConfig);

    createUserConfig(basePath, {
      migrate: {
        setup: {
          lint: { command: 'touch', args: ['custom-lint-config-script'] },
        },
      },
    });

    const options = createMigrateOptions(basePath, { userConfig: 'rehearsal-config.json' });
    const userConfig = new UserConfig(basePath, 'rehearsal-config.json', 'migrate');
    const tasks = [lintConfigTask(options, { userConfig })];
    await listrTaskRunner(tasks);

    // This proves the custom command works
    expect(readdirSync(basePath)).not.toContain('custom-lint-config-script');
    expect(output).toContain('[SKIPPED] Create eslint config');
  });
});
