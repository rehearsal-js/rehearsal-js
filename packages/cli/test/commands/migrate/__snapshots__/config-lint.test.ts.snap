// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task: config-lint > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;
